rclcpp/src/rclcpp/detail/mutex_two_priorities.cpp

```diff
@@ -47,16 +68,23 @@ LowPriorityLockable::LowPriorityLockable(MutexTwoPriorities & parent)
 void
 LowPriorityLockable::lock()
 {
-  std::unique_lock<std::mutex> barrier_guard{parent_.barrier_};
-  parent_.data_.lock();
-  barrier_guard.release();
+  std::unique_lock<std::mutex> guard{parent_.cv_mutex_};
+  while (parent_.data_taken_ || parent_.hp_waiting_count_) {
```

Comment:
```suggestion
  while (parent_.hp_waiting_count_ || parent_.data_taken_) {
```

high priority should be checked 1st? since that means that there is at least one high priority thread in wait.

Reply:
The order is irrelevant.
These are two boolean flags, so short-circuit behavior doesn't matter.